lib/uninterleaveAndPackBuffers.js

```diff
@@ -24,148 +24,122 @@ module.exports = uninterleaveAndPackBuffers;
-        pipeline = {};
-        extras._pipeline = pipeline;
-    }
-    return pipeline;
-}
+    });
 
-function packAccessors(gltf, accessorsToPack, sourceBuffer, packBuffer, bufferViewId, originalOffset, packOffset) {
-    var accessors = gltf.accessors;
-    var bufferViews = gltf.bufferViews;
-    var length = accessorsToPack.length;
-    var bytesWritten = 0;
-    for (var i = 0; i < length; i++) {
-        var accessorId = accessorsToPack[i];
-        var accessor = accessors[accessorId];
-        var bufferView = bufferViews[accessor.bufferView];
-        var byteStride = getAccessorByteStride(accessor);
-        var byteOffset = accessor.byteOffset + bufferView.byteOffset;
-        var numberOfComponents = numberOfComponentsForType(accessor.type);
-        var componentByteLength = byteLengthForComponentType(accessor.componentType);
-        var byteLength = numberOfComponents * componentByteLength;
-        var offset = byteOffset;
-        var count = accessor.count;
-        accessor.byteStride = 0;
-        // Store as temp for now, finalize later
-        var extras = getPipelineExtras(accessor);
-        extras.bufferView = bufferViewId;
-        accessor.byteOffset = packOffset - originalOffset + bytesWritten;
-        for (var num = 0; num < count; num++) {
-            for (var j = 0; j < byteLength; j++) {
-                packBuffer[packOffset + bytesWritten] = sourceBuffer[offset + j];
-                bytesWritten++;
+    var buffers = [{
+        byteLength: bufferData.length,
+        extras: {
+            _pipeline: {
+                source: bufferData,
+                extension: '.bin'
             }
-            offset += byteStride;
         }
-    }
-    return packOffset + bytesWritten;
-}
+    }];
 
-function getAccessorsByTargetAndByteLength(gltf, bufferId) {
-    var accessors = gltf.accessors;
-    var bufferViews = gltf.bufferViews;
-    var accessorsByTargetAndByteLength = {};
-    for (var accessorId in accessors) {
-        if (accessors.hasOwnProperty(accessorId)) {
-            var accessor = accessors[accessorId];
-            var bufferView = bufferViews[accessor.bufferView];
-            if (bufferView.buffer === bufferId) {
-                var target = defaultValue(bufferView.target, 0);
-                var accessorsByByteLength = accessorsByTargetAndByteLength[target];
-                if (!defined(accessorsByByteLength)) {
-                    accessorsByByteLength = {};
-                    accessorsByTargetAndByteLength[target] = accessorsByByteLength;
-                }
-                var byteLength = byteLengthForComponentType(accessor.componentType);
-                var accessorIds = accessorsByByteLength[byteLength];
-                if (!defined(accessorIds)) {
-                    accessorIds = [];
-                    accessorsByByteLength[byteLength] = accessorIds;
-                }
-                accessorIds.push(accessorId);
-            }
-        }
+    var bufferViews = [{
```

Comment:
In this file, it packs all the bufferViews if they are belong to an accessor and write over the original array of bufferViews. However, if there are some bufferViews are not belong to any accessor, e.g. is a property of an extension, then it won't work.